hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/PluralAttributeMappingImpl.java

```diff
@@ -382,14 +382,6 @@ public Fetch generateFetch(
 									creationState.getSqlAstCreationState().getSqlExpressionResolver(),
 									creationState.getSqlAstCreationState().getCreationContext()
 							);
-
-							lhsTableGroup.addTableGroupJoin( tableGroupJoin );
-
-							sqlAstCreationState.getFromClauseAccess().registerTableGroup(
-									fetchablePath,
-									tableGroupJoin.getJoinedGroup()
-							);
-
```

Comment:
@dreab8 the above code seems duplicated. The first one has been done in the constructor above, and the second one will be done in the external method. What do you think? Duplicating concern will confuse code reader and end up with potential maintenance difficulty.

Reply:
the code should be removed, it was a mistake to add this code, good catch